master/buildbot/reporters/syslogreporter.py

```diff
@@ -0,0 +1,128 @@
+import syslog
+
+from buildbot import config
+from buildbot.reporters import utils
+from buildbot.util import service
+from buildbot.process.results import CANCELLED
+from buildbot.process.results import EXCEPTION
+from buildbot.process.results import FAILURE
+from buildbot.process.results import SUCCESS
+from buildbot.process.results import WARNINGS
+
+
+from twisted.internet import defer
+
+
+class SyslogStatusPush(service.BuildbotService):
+    neededDetails = dict()
+    name = "SyslogStatusPush"
+
+    def _checkLogSeverity(self, logSeverity):
+        if logSeverity not in [syslog.LOG_EMERG, syslog.LOG_ALERT,
+                               syslog.LOG_CRIT, syslog.LOG_ERR,
+                               syslog.LOG_WARNING, syslog.LOG_NOTICE,
+                               syslog.LOG_INFO, syslog.LOG_DEBUG]:
+            config.error("Please provide a valid log severity")
+
+    def checkConfig(self, tags=None, builders=None, wantLogs=True,
+                    buildSetSummary=False, identifier="buildbot",
+                    cancelledSeverity=syslog.LOG_INFO,
+                    exceptionSeverity=syslog.LOG_INFO,
+                    failureSeverity=syslog.LOG_INFO,
+                    successSeverity=syslog.LOG_INFO,
+                    warningsSeverity=syslog.LOG_INFO):
+        service.BuildbotService.checkConfig(self)
+        if builders is not None and tags is not None:
+            config.error("Please specify only builders or tags to include - " +
+                         "not both.")
+
+        for severity in [cancelledSeverity, exceptionSeverity, failureSeverity,
+                         successSeverity, warningsSeverity]:
+            self._checkLogSeverity(severity)
+
+    def reconfigService(self, tags=None, builders=None, wantLogs=True,
+                        buildSetSummary=False, identifier="buildbot",
+                        cancelledSeverity=syslog.LOG_INFO,
+                        exceptionSeverity=syslog.LOG_INFO,
+                        failureSeverity=syslog.LOG_INFO,
+                        successSeverity=syslog.LOG_INFO,
+                        warningsSeverity=syslog.LOG_INFO):
+        # If Tags and Builders are both None it will take all builds.
+        self.tags = tags
+        self.builders = builders
+        self.wantLogs = wantLogs
```

Comment:
Maybe you should consider using a method similar to the [http reporter](https://github.com/buildbot/buildbot/blob/master/master/buildbot/reporters/http.py#L49-L51) to allow the user to specify what extra data they want to pull for their builds and buildsets.